scripts/testnet.sh

```diff
@@ -157,13 +194,20 @@ update_if_existing() {
 # $1=vm name
 # $2=docker image
 # $3=environment id
+# $4=preemptible
 # NB: --run needs to be at the end or it will ignore the other arguments.
 start_testnode_vm() {
   local vm_name=${1}
   local docker_image=${2}
   local environment_id=${3}
   local api_token="${HOPRD_API_TOKEN}"
   local password="${BS_PASSWORD}"
+  local preemptible_args=""
+  local preemptible="${4:-}"
+
+  if [ -n "${preemptible}" ]; then
+    preemptible_args="--preemptible --metadata-from-file shutdown-script=scripts/gcloud-shutdown.sh"
```

Comment:
Does gcloud upload that script somewhere?

Reply:
It stores it as a value in the VM metadata key-value section